x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/RoleDescriptorStore.java

```diff
@@ -139,7 +139,7 @@ public void resolveCrossClusterAccessRoleReference(
     ) {
         final Set<RoleDescriptor> roleDescriptors = crossClusterAccessRoleReference.getRoleDescriptorsBytes().toRoleDescriptors();
         for (RoleDescriptor roleDescriptor : roleDescriptors) {
-            if (hasPrivilegesOtherThanIndex(roleDescriptor)) {
+            if (roleDescriptor.hasPrivilegesOtherThanIndex()) {
```

Comment:
Sorry, I messed up addressing a [comment](https://github.com/elastic/elasticsearch/pull/94968#discussion_r1157897658) in the previous PR. Addressing it here.